src/dex-ui-components/presets/AddLiquidityForm/AddLiquidityForm.tsx

```diff
@@ -115,28 +115,57 @@ export function AddLiquidityForm(props: AddLiquidityFormProps) {
     tokenToReceiveSymbol: formValues.secondToken.symbol,
   });
 
+  const allPoolFee = getAllPoolTransactionFee({
+    tokenPairs: props.tokenPairs,
+    poolSymbol: formValues.poolName,
+  });
+
   const tokensWithPairs = getTokensByUniqueAccountIds(props.tokenPairs ?? []);
   const tokensPairedWithFirstToken = getPairedTokens(
     formValues.firstToken?.tokenMeta?.tokenId ?? "",
-    formValues.firstToken?.tokenMeta?.pairAccountId ?? "",
     props.tokenPairs ?? []
   );
 
-  /** Update Balances */
-  useEffect(() => {
-    const firstTokenBalance = getTokenBalance(
-      formValues.firstToken.tokenMeta.tokenId ?? "",
-      props.pairedAccountBalance
-    );
-    const secondTokenBalance = getTokenBalance(
-      formValues.secondToken.tokenMeta.tokenId ?? "",
-      props.pairedAccountBalance
-    );
+  const updateBalances = (firstTokenId: string, secondTokenId: string) => {
+    const firstTokenBalance = getTokenBalance(firstTokenId, props.pairedAccountBalance);
+    const secondTokenBalance = getTokenBalance(secondTokenId, props.pairedAccountBalance);
     addLiquidityForm.setValue("firstToken.balance", firstTokenBalance);
     addLiquidityForm.setValue("secondToken.balance", secondTokenBalance);
+  };
+
+  /** Update Balances */
+  useEffect(() => {
+    updateBalances(formValues.firstToken.tokenMeta.tokenId ?? "", formValues.secondToken.tokenMeta.tokenId ?? "");
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [JSON.stringify(props.pairedAccountBalance)]);
 
+  useEffect(() => {
+    const selectedPair = props.tokenPairs?.find(
+      (tokenPair) => tokenPair.tokenA.tokenMeta.pairAccountId === props.selectedFromPoolPairId
+    );
+    if (selectedPair) {
+      addLiquidityForm.setValue("firstToken", selectedPair.tokenA);
+      addLiquidityForm.setValue("secondToken", selectedPair.tokenB);
+      addLiquidityForm.setValue("fee", selectedPair.tokenA.tokenMeta.fee?.toNumber() ?? 0);
+      addLiquidityForm.setValue("poolName", props.poolName ?? "");
+      updateBalances(selectedPair.tokenA.tokenMeta.tokenId ?? "", selectedPair.tokenB.tokenMeta.tokenId ?? "");
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [props.selectedFromPoolPairId, props.tokenPairs]);
+
+  function handleTransactionFeeClicked(event: ChangeEvent<HTMLInputElement>) {
```

Comment:
handleTransactionFee`Changed` is more exact since this captures a change event.

Reply:
Done